pkg/standalone-utils/contracts/ProtocolFeeHelper.sol

```diff
@@ -9,60 +9,24 @@ import { IProtocolFeeController } from "@balancer-labs/v3-interfaces/contracts/v
 import { IVaultErrors } from "@balancer-labs/v3-interfaces/contracts/vault/IVaultErrors.sol";
 import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
 
-import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
+import { PoolHelperCommon } from "./PoolHelperCommon.sol";
 
-contract ProtocolFeeHelper is IProtocolFeeHelper, SingletonAuthentication {
+contract ProtocolFeeHelper is IProtocolFeeHelper, PoolHelperCommon {
     using EnumerableSet for EnumerableSet.AddressSet;
 
-    EnumerableSet.AddressSet private _pools;
-
     modifier withKnownPool(address pool) {
```

Comment:
```suggestion
    modifier withPoolAdded(address pool) {
```

Reply:
I don't know... maybe this one's ok as is. "withPoolAdded" sounds weird, even though it would make the name more consistent. Maybe withValidPool? "Add" ensures it is registered - plus passes any additional validations specific to that helper - so this means it's been validated and added to the set before use.

Reply:
I don't have a strong opinion; it's just that I got used to naming things 'consistently' by sticking to the same word even if it sounds dumb as opposed to calling the same thing by two names.

`addPool` adds a pool, so the condition to check is whether the pool was added, not registered / valid --> `withPoolAdded` (or `withAddedPool`?). It doesn't sound that nice, but it sticks to the same word and suggests that you first need to call `addPool` to call the function affected by the modifier in a very direct manner.

I'm also fine with this option though; the contract is not complex enough to cause that much confusion around this/